src/libcore/cell.rs

```diff
@@ -280,6 +281,19 @@ unsafe impl<T> Send for Cell<T> where T: Send {}
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<T> !Sync for Cell<T> {}
 
+#[unstable(feature = "move_cell", issue = "39264")]
+impl<T:Clone> Clone for Cell<T> {
+    #[inline]
+    default fn clone(&self) -> Cell<T> {
+        let temp = unsafe { mem::uninitialized() };
+        let inner = self.replace(temp);
+        // If `clone` panics, let it crash.
```

Comment:
Yes. This is very unsafe. But since the `Cell` dose not satisfy `UnwindSafe` trait, it can't be used in `catch_unwind`, so this behavior can't be observed by users.

The unsafety only occurs if users try to use `AssertUnwindSafe` wraps a `Cell` explicitly. But this is the user's fault.